network/message/authorization.go

```diff
@@ -0,0 +1,364 @@
+		},
+		Config: map[channels.Channel]flow.RoleList{
+			channels.PushTransactions: {flow.RoleCollection}, // channel alias ReceiveTransactions = PushTransactions
+		},
+	}
+
+	// core messages for execution & verification
+	authorizationConfigs[ExecutionReceipt] = MsgAuthConfig{
+		Name: ExecutionReceipt,
+		Type: func() interface{} {
+			return new(flow.ExecutionReceipt)
+		},
+		Config: map[channels.Channel]flow.RoleList{
+			channels.PushReceipts: {flow.RoleExecution}, // channel alias ReceiveReceipts = PushReceipts
+		},
+	}
+	authorizationConfigs[ResultApproval] = MsgAuthConfig{
+		Name: ResultApproval,
+		Type: func() interface{} {
+			return new(flow.ResultApproval)
+		},
+		Config: map[channels.Channel]flow.RoleList{
+			channels.PushApprovals: {flow.RoleVerification}, // channel alias ReceiveApprovals = PushApprovals
+		},
+	}
+
+	// data exchange for execution of blocks
+	authorizationConfigs[ChunkDataRequest] = MsgAuthConfig{
+		Name: ChunkDataRequest,
+		Type: func() interface{} {
+			return new(messages.ChunkDataRequest)
+		},
+		Config: map[channels.Channel]flow.RoleList{
+			channels.RequestChunks: {flow.RoleVerification}, // channel alias RequestChunks = ProvideChunks
+		},
+	}
+	authorizationConfigs[ChunkDataResponse] = MsgAuthConfig{
+		Name: ChunkDataResponse,
+		Type: func() interface{} {
+			return new(messages.ChunkDataResponse)
+		},
+		Config: map[channels.Channel]flow.RoleList{
+			channels.ProvideChunks: {flow.RoleExecution}, // channel alias RequestChunks = ProvideChunks
+		},
+	}
+
+	// result approvals
+	authorizationConfigs[ApprovalRequest] = MsgAuthConfig{
+		Name: ApprovalRequest,
+		Type: func() interface{} {
+			return new(messages.ApprovalRequest)
+		},
+		Config: map[channels.Channel]flow.RoleList{
+			channels.RequestApprovalsByChunk: {flow.RoleConsensus}, // channel alias ProvideApprovalsByChunk  = RequestApprovalsByChunk
+		},
+	}
+	authorizationConfigs[ApprovalResponse] = MsgAuthConfig{
+		Name: ApprovalResponse,
+		Type: func() interface{} {
+			return new(messages.ApprovalResponse)
+		},
+		Config: map[channels.Channel]flow.RoleList{
+			channels.ProvideApprovalsByChunk: {flow.RoleVerification}, // channel alias ProvideApprovalsByChunk  = RequestApprovalsByChunk
+
+		},
+	}
+
+	// generic entity exchange engines
+	authorizationConfigs[EntityRequest] = MsgAuthConfig{
+		Name: EntityRequest,
+		Type: func() interface{} {
+			return new(messages.EntityRequest)
+		},
+		Config: map[channels.Channel]flow.RoleList{
+			channels.RequestReceiptsByBlockID: {flow.RoleConsensus},
+			channels.RequestCollections:       {flow.RoleAccess, flow.RoleExecution},
+		},
+	}
+	authorizationConfigs[EntityResponse] = MsgAuthConfig{
+		Name: EntityResponse,
+		Type: func() interface{} {
+			return new(messages.EntityResponse)
+		},
+		Config: map[channels.Channel]flow.RoleList{
+			channels.ProvideApprovalsByChunk:  {flow.RoleVerification},
+			channels.ProvideReceiptsByBlockID: {flow.RoleExecution},
```

Comment:
I am wondering if this maybe should be 
```golang
channels.ProvideReceiptsByBlockID: {flow.RoleExecution},
channels.ProvideCollections:       {flow.RoleCollection},
```

* Based on my (arguably very high-level understanding), requesting approvals and respective responses go through the channel `RequestApprovalsByChunk` (alias `ProvideApprovalsByChunk`); which is specified right above https://github.com/onflow/flow-go/blob/78e0702c2f401c7030a73d3de055dc01d0d3b6c9/network/message/authorization.go#L206-L225
* we specify that collections can be requested through the `RequestCollections` channel (implemented as an `EntityRequest`): https://github.com/onflow/flow-go/blob/78e0702c2f401c7030a73d3de055dc01d0d3b6c9/network/message/authorization.go#L235 But we don't specify any means for collectors to answer with an `EntityResponse`

Reply:
👍 Agreed, I think we're missing `channels.ProvideCollections:       {flow.RoleCollection},` here

Reply:
https://github.com/onflow/flow-go/pull/2648/commits/791c6b9bc217050cd9925f12a24bf7fb35f96529